GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -16,22 +16,23 @@ public class AheadBehindDataProvider : IAheadBehindDataProvider
     {
         private readonly Func<IExecutable> _getGitExecutable;
 
+        // TODO handle [gone] status to show that remote branch no longer exists
         private readonly Regex _aheadBehindRegEx =
-            new Regex(@"^\[(ahead (?<ahead>\d+))?(, )?(behind (?<behind>\d+))?\] (?<branch>.*)$",
+            new Regex(@"^(\[(ahead (?<ahead_u>\d+))?(, )?(behind (?<behind_u>\d+))?\])?::(\[(ahead (?<ahead_p>\d+))?(, )?(behind (?<behind_p>\d+))?\])?::(?<branch>.*)$",
```

Comment:
Can we please have tests where we get to exercise the full regex and each the left and the right groups?